packages/data-stores/src/plans/types.ts

```diff
@@ -63,6 +63,7 @@ export interface PlanProduct {
 export interface PlanIntroductoryOffer {
 	formattedPrice: string;
 	rawPrice: number;
+	rawPriceInteger: number;
```

Comment:
Just to confirm /fyi - this will always be sent along the `rawPrice`.  We make this assumption with all of these props, that if an intro offer exists then they are all populated /sent through from endpoint. It's probably something to revisit at some point and assume underfined/optional props

p.s. we could also only send or unpack the `rawPriceInteger`. I don't think we need both or have cases where the non-integer one is preferred, but could be wrong.

Reply:
I believe, ideally, we have similar format for all prices: original, discounted, and introductory offer.

For original and discounted there are full and monthly prices, both in integer.

> p.s. we could also only send or unpack the rawPriceInteger. I don't think we need both or have cases where the non-integer one is preferred, but could be wrong.
I think it just require significantly more work to adjust all existing usages.

Reply:
Yes, all of the pricing properties in the data-store are integer-based. We only map/use `raw_price_integer` and `orig_cost_integer` from the `usePlans`/`useSitePlans` queries (and hence the `usePricingMetaForGridPlans` hook).

> I think it just require significantly more work to adjust all existing usages.

Maybe. As far as the data-store hooks go, I think there's only one location where we use the intro-offer price in the plans grid ([header-price.tsx](https://github.com/Automattic/wp-calypso/blob/trunk/packages/plans-grid-next/src/components/header-price.tsx)). I'm not sure if used anywhere else outside the plans-grid, but we would need to verify.

Reply:
p.s. maybe unrelated, but in case it's relevant to the work -> the way we did the intro-offers for WPCOM previously (you can read more about the work in PT: pau2Xa-5mX-p2 ) is we changed the plan prices to reflect the "upgrade" price, then used the intro offer price for the main price display (with `isSmallestUnit={ false }` for `PlanPrice`). So the display was (as shown in https://github.com/Automattic/wp-calypso/pull/84763 ):

<img width="400" alt="Screenshot 2024-07-07 at 11 50 07 AM" src="https://github.com/Automattic/wp-calypso/assets/1705499/3beb45a7-3eaf-4ed3-a7bf-a5a75d623d7f">

so for premium: 
- intro-offer: $8
- premium regular price updated to: $10